packages/vite/src/node/server/index.ts

```diff
@@ -175,7 +175,7 @@ export interface ViteDevServer {
   /**
    * web socket server with `send(payload)` method
    */
-  ws: WebSocketServer
+  ws?: WebSocketServer
```

Comment:
This could be a breaking change as there are quite a few plugins relying on it. Maybe it's better to have a mock ws instance instead of `undefined`

Reply:
Thinking this further on, the proxy could log e.g. debug messages that the object xy would be sended, but the proxy/mock is used